### PR TITLE
Replace axios by fetch, and simplify

### PR DESCRIPTION
**Pull request by caub**
Replaced axios by fetch (more standard and simple), and simplified the rest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +61,8 @@
     ],
     "files": [
       "src/**/*.test.ts"
-    ]
+    ],
+    "serial": true
```
Comment (caub) [inline package.json]: when using sinon.stub's on an async function like `client.post` it's important to run test serially

**Comment by linyows** [comment]
This library is supposed to work on the browser. Does node-fetch work correctly in the browser?

**Comment by linyows** [comment]
What is the motivation for paying for the library that is working properly now?

**Comment by caub** [comment]
Hey @linyows, node-fetch is working also when bundling for browsers (https://github.com/bitinn/node-fetch/blob/master/package.json#L6)
The reason for moving to fetch is because it's a standard, and simpler API

> What is the motivation for paying for the library

What do you mean? I just contribute, I know it's working currently

**Comment by linyows** [comment]
@caub Thank you for your reply. We use this library in a production environment. There are not many similar cases other than us. I understand that using the node-fetch slightly reduces the number of rows. However, I do not think that the code of the master is complicated now, nor do I think that there is a problem with axios. On the other hand, the part using Generics for mixin has a lot of learning.

Can you change to refactoring with Generics only?

**Comment by caub** [comment]
@linyows so I tried isolating the mixins here: https://github.com/pepabo/gmopg/pull/13

and on that branch I've put back axios

You choose, personally I prefer this heavier one, less tied to axios (node-fetch is completely production ready btw)

**Comment by linyows** [comment]
@caub Now that you have split P/R, we are ready to merge the major changes that will change the library. Thanks. There are tiny conflict. I will merge after this resolved.

**Comment by caub** [comment]
Yes, thanks, I've tried to do it

**Comment by linyows** [comment]
Many thanks!

**Pull request merged**